app/src/main/java/org/p2p/wallet/moonpay/ui/new/BuyPresenter.kt

```diff
@@ -230,25 +231,35 @@ class NewBuyPresenter(
     }
 
     private fun isValidCurrencyForPay(): Boolean {
-        val selectedCurrencyCode = selectedCurrency.abbreviation
         val currentPaymentMethod = selectedPaymentMethod ?: return false
 
-        if (currentPaymentMethod.method == PaymentMethod.MethodType.BANK_TRANSFER) {
-            if (selectedCurrencyCode == Constants.USD_READABLE_SYMBOL ||
-                (currentAlpha3Code == BANK_TRANSFER_UK_CODE && selectedCurrencyCode == EUR_READABLE_SYMBOL)
-            ) {
-                paymentMethods.find { it.method == PaymentMethod.MethodType.CARD }?.let {
-                    onPaymentMethodSelected(it, byUser = false)
-                }
-                return isValidCurrencyForPay()
-            } else if (selectedCurrency == FiatCurrency.GBP && currentAlpha3Code != BANK_TRANSFER_UK_CODE) {
-                paymentMethods.find { it.method == PaymentMethod.MethodType.CARD }?.let {
-                    onPaymentMethodSelected(it, byUser = false)
-                }
-                return isValidCurrencyForPay()
+        if (currentPaymentMethod.method != PaymentMethod.MethodType.BANK_TRANSFER) {
+            return true
+        }
+
+        val isUsdSelected =
+            selectedCurrency == FiatCurrency.USD
+        val isEuValidCurrencySelected =
+            currentAlpha3Code == BANK_TRANSFER_UK_CODE && selectedCurrency == FiatCurrency.EUR
+
+        return when {
+            isUsdSelected || isEuValidCurrencySelected -> {
+                switchToCardPaymentMethod()
+                isValidCurrencyForPay()
+            }
+            selectedCurrency == FiatCurrency.GBP && currentAlpha3Code != BANK_TRANSFER_UK_CODE -> {
+                switchToCardPaymentMethod()
+                isValidCurrencyForPay()
+            }
```

Comment:
duplication, I suggest using simple if instead avoid confusions about different states, something like this:
```kotlin
if (currentPaymentMethod.method != PaymentMethod.MethodType.BANK_TRANSFER) {
    return true
}

val isUsdSelected = selectedCurrency == FiatCurrency.USD
val isEuValidCurrencySelected = currentAlpha3Code == BANK_TRANSFER_UK_CODE && selectedCurrency == FiatCurrency.EUR
val isGbpWithDifferentCode = selectedCurrency == FiatCurrency.GBP && currentAlpha3Code != BANK_TRANSFER_UK_CODE

if (isUsdSelected || isEuValidCurrencySelected || isGbpWithDifferentCode) {
    switchToCardPaymentMethod()
    return isValidCurrencyForPay()
}

return true
```

Reply:
yeaah, sounds good